src/python/pants/reporting/zipkin_reporter.py

```diff
@@ -191,3 +204,129 @@ def bulk_record_workunits(self, engine_workunits):
 
 def from_secs_and_nanos_to_float(secs, nanos):
   return secs + ( nanos / NANOSECONDS_PER_SECOND )
+
+
+def emit_spans(self):
```

Comment:
Can you add some comments explaining why this exists, what it's mostly copied from, and what changes you've made to it?